crates/context/primitives/src/client.rs

```diff
@@ -46,37 +59,232 @@ impl ContextClient {
         Ok(Some(context))
     }
 
-    pub async fn get_contexts(&self) -> impl Stream<Item = eyre::Result<ContextId>> {
-        stream::empty()
+    pub async fn get_contexts(
+        &self,
+        start: Option<ContextId>,
+    ) -> impl Stream<Item = eyre::Result<ContextId>> {
+        let datastore = self.datastore.clone();
+
+        try_stream! {
+            let handle = datastore.handle();
+            let iter_result = handle.iter::<ContextMetaKey>();
+            let mut iter = iter_result?;
+
+            let start = start.and_then(|s| iter.seek(ContextMetaKey::new(s)).transpose());
+
+            for key in start.into_iter().chain(iter.keys()) {
+                yield key?.context_id();
+            }
+        }
     }
 
     pub async fn delete_context(&self, context_id: &ContextId) -> eyre::Result<bool> {
-        todo!()
+        let mut handle = self.datastore.handle();
+
+        let key = ContextMetaKey::new(*context_id);
+        if !handle.has(&key)? {
+            return Ok(false);
+        }
+
+        handle.delete(&key)?;
+
+        handle.delete(&ContextConfigKey::new(*context_id))?;
+
+        let identity_keys = {
+            let mut iter = handle.iter::<ContextIdentityKey>()?;
+            iter.keys()
+                .filter_map(Result::ok)
+                .filter(|k| k.context_id() == *context_id)
+                .collect::<Vec<_>>()
+        };
+
+        for key in identity_keys {
+            handle.delete(&key)?;
+        }
+
+        Ok(true)
     }
 
     pub async fn context_members(
         &self,
         context_id: &ContextId,
         only_owned: Option<bool>,
     ) -> impl Stream<Item = eyre::Result<PublicKey>> {
-        stream::empty()
+        let datastore = self.datastore.clone();
+        let context_id = *context_id;
+        let only_owned = only_owned.unwrap_or(false);
+
+        try_stream! {
+            let handle = datastore.handle();
```

Comment:
same thing here, move the handle out, it does `.clone()` internally

https://github.com/calimero-network/core/blob/24f0cc6bfea9476314bf57784a503cee05d1d0a5/crates/store/src/lib.rs#L34-L36